kong/pdk/service/request.lua

```diff
@@ -284,6 +284,14 @@ local function new(self)
       ngx.var.upstream_host = value
     end
 
+    if string_lower(header) == ":authority"
+       and (ngx.var.upstream_scheme == "grpc" or
+           ngx.var.upstream_scheme == "grpcs") then
+
+      local resty_grpc = require "resty.kong.grpc"
+      return resty_grpc.set_authority(value)
```

Comment:
This function can be cached at line 263 above, as so we do not load it every time the function is used.